src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -769,11 +769,13 @@ async function fetchFeatures(params: { extensionPath: string; cwd: string; outpu
 					throw new Error(err);
 				}
 
-				if (!(await applyFeatureConfigToFeature(output, featureSet, feature, featCachePath))) {
+				let updatedFeature = await applyFeatureConfigToFeature(output, featureSet, feature, featCachePath);
+				if (!updatedFeature) {
 					const err = `Failed to parse feature '${featureDebugId}'. Please check your devcontainer.json 'features' attribute.`;
 					throw new Error(err);
 				}
 
+				featuresConfig.featureSets[i].features[0] = updatedFeature;
```

Comment:
Unless you noticed otherwise, it should be fine to just set this value in `appleFeatureConfigToFeature`. The featureConfig is passed as a ["copy of a reference"](https://stackoverflow.com/questions/13104494/does-javascript-pass-by-reference)

Reply:
that would prevent the need to switch to a counted for loop as well

Reply:
Ah! I did not realize that featureConfig is passed as a ["copy of a reference"](https://stackoverflow.com/questions/13104494/does-javascript-pass-by-reference). 

Modified my changes